rpcs3/Emu/RSX/CgBinaryFragmentProgram.cpp

```diff
@@ -212,10 +211,10 @@ template<typename T> std::string CgBinaryDisasm::GetSrcDisAsm(T src)
 	swizzle += f[src.swizzle_z];
 	swizzle += f[src.swizzle_w];
 
-	if (swizzle == ".xxxx") swizzle = ".x";
```

Comment:
FTR, std::string has specialized handling for null terminated sequence of char (i.e CString) making all the sv stuff pointless.

Reply:
Same goes for all variants of std::basic_string<T>, you can always compare with a null-terminated sequence of T with an optimized code path.

Reply:
Thats far less optimized actually, because with string view it can compare the size of strings (most of comparisons failures end here, making ut super fast for it) and then executing multiple bytes comparisons at once with SSE/AVX (or just u64/u32/u15l6 loads) without worrying about "reading invalid bytes after null terminator".